### DISCUSSION TITLE
ValueError: The size of the proposed random crop ROI is larger than the image size.

### DISCUSSION THREAD

**Discussion by talhaanwarch** [Project-MONAI/MONAI#2593] [Q&A]
While using 
```
RandCropByPosNegLabeld(
            keys=["image", "label"],
            label_key="label",
            spatial_size=(96, 96, 96),
            pos=1,
            neg=0.5,
            num_samples=4,
            image_key="image",
            image_threshold=0,
        ),
```
I am getting this error. What parameters should I change to avoid this? like do I need to choose a larger spatial size? Can you also explain what is pos and neg. is it foreground and background?

**Comment by Nic-Ma** [comment]
Hi @talhaanwarch ,

I think you missed to attach your error.
BTW, pos and neg map to foreground and background.

Thanks.

**Reply by talhaanwarch** [reply]
Thanks @Nic-Ma , this is the error. `ValueError: The size of the proposed random crop ROI is larger than the image size.` I resolved the error by decreasing spatial_size to (64,64,64), is it the correct approach? or what else I can do other than reducing spatial size.

**Reply by Nic-Ma** [reply]
Yes, it's the correct approach because your previous ROI is too big, if your dataset has different image shape, maybe you can resize them or pad to the same size first.
Thanks.

**Reply by talhaanwarch** [reply]
I already tried that, but z-axis ranges from 30 to 1000, and padding all images to 1000 to have equal shape is quite expensive. I also noted that notebooks in the tutorial do not use this transform in validation data, but I have to use this, otherwise, I will get dimension error. is it ok to use it?

**Reply by Nic-Ma** [reply]
OK, maybe @rijobro can help share more advice on the training parameter tuning?

Thanks in advance.

**Reply by talhaanwarch** [reply]
thanks a lot, I am really having issue in transform selection

**Reply by xiefeier** [reply]
Dimension Error can be resolved by modifying the parser parameters roi_x, roi_y, roi_z in __main__.py

**Reply by rijobro** [reply]
You could use `SpatialPadd` to ensure your images are all at least `(96, 96, 96)`. This will convert any images with z-size=30 to z-size=96. Any images that were originally bigger than this (e.g., z-size=1000) will remain unchanged. Once all of your images are at least equal to the size of your random crop `(96, 96, 96)`, you can then safely use `RandCropByPosNegLabeld`.
